tests/test_financial.py

```diff
@@ -164,27 +165,48 @@ def test_rate_maximum_iterations_exception_array(self):
 class TestNpv:
     def test_npv(self):
         assert_almost_equal(
-            npf.npv(0.05, [-15000, 1500, 2500, 3500, 4500, 6000]),
+            npf.npv(0.05, [-15000.0, 1500.0, 2500.0, 3500.0, 4500.0, 6000.0]),
             122.89, 2)
 
     def test_npv_decimal(self):
         assert_equal(
             npf.npv(Decimal('0.05'), [-15000, 1500, 2500, 3500, 4500, 6000]),
             Decimal('122.894854950942692161628715'))
 
-    def test_npv_broadcast(self):
-        cashflows = [
+    def test_npv_broadcast_cashflows(self):
+        cashflows = np.array([
             [-15000, 1500, 2500, 3500, 4500, 6000],
             [-15000, 1500, 2500, 3500, 4500, 6000],
             [-15000, 1500, 2500, 3500, 4500, 6000],
             [-15000, 1500, 2500, 3500, 4500, 6000],
-        ]
-        expected_npvs = [
-            122.8948549, 122.8948549, 122.8948549, 122.8948549
-        ]
+        ])
+        expected_npvs = numpy.array([
+            [122.8948549, 122.8948549, 122.8948549, 122.8948549]
+        ])
         actual_npvs = npf.npv(0.05, cashflows)
         assert_allclose(actual_npvs, expected_npvs)
 
+    @pytest.mark.parametrize("dtype", [Decimal, float])
+    def test_npv_broadcast_equals_for_loop(self, dtype):
+        cashflows_str = [
+            ["-15000.0", "1500.0", "2500.0", "3500.0", "4500.0", "6000.0"],
+            ["-25000.0", "1500.0", "2500.0", "3500.0", "4500.0", "6000.0"],
+            ["-35000.0", "1500.0", "2500.0", "3500.0", "4500.0", "6000.0"],
+            ["-45000.0", "1500.0", "2500.0", "3500.0", "4500.0", "6000.0"],
+        ]
+        rates_str = ["-0.05", "0.00", "0.05", "0.10", "0.15"]
+
+        cashflows = np.array([[dtype(x) for x in cf] for cf in cashflows_str])
+        rates = np.array([dtype(x) for x in rates_str])
+
+        expected = np.empty((len(rates), len(cashflows)), dtype=dtype)
+        for i, r in enumerate(rates):
+            for j, cf in enumerate(cashflows):
+                expected[i, j] = npf.npv(r, cf).item()
```

Comment:
Here we're using a for loop to calculate the equivalent to the "broadcasting" operation. This is slow, however I am more confident that it is correct.